main/src/library/Map.flix

```diff
@@ -51,6 +51,22 @@ instance Order[Map[k, v]] with Order[k], Order[v] {
 
 }
 
+instance Functor[Map[k]] {
+    pub def map(f: v1 -> v2 & ef, m: Map[k, v1]): Map[k, v2] & ef =
+        let Map(tree) = m;
+        Map(RedBlackTree.mapWithKey((_, v) -> f(v), tree))
+}
+
+instance Foldable[Map[k]] {
+    pub def foldLeft(f: (b, v) -> b & ef, s: b, m: Map[k, v]): b & ef = Map.foldLeft((acc, v) -> f(acc, v), s, m)
+    pub def foldRight(f: (v, b) -> b & ef, s: b, m: Map[k, v]): b & ef = Map.foldRight((v, acc) -> f(v, acc), s, m)
+}
+
+instance Traversable[Map[k]] {
+    pub def traverse(f: a -> m[b] & ef, t: Map[k, a]): m[Map[k, b]] & ef with Applicative[m] = Map.traverse(f, t)
+    pub override def sequence(t: Map[k, m[a]]): m[Map[k, a]] with Applicative[m] = Map.sequence(t)
+}
+
```

Comment:
All of the above instances allow me to map/fold/traverse on the value part of a map, right? Keeping the keys the same?

Reply:
Yes - keys are kept the same (altering could change the "shape" of the map)